Feature / REL-1170 / Migration from sql to neo4j

vbyndych opened the pull request:
## Goal
As a part of an effort to develop RDF persistence in neo4j, we need a procedure to migrate RDF data from SQL database (MySQL or PgSQL) to Neo4j.

## Changelog
- fix: RDF migration from sql to neo4j 
- fix: code style for legacy class.

## How to test / use

Internal use only.

vbyndych committed:
fix: RDF migration from sql to neo4j

- fixed typo which causes import to fail
- created migratino to update DB records properly.

vbyndych committed:
fix: code style for legacy class.

augustas commented on migrations/Version202307041134591871_taoTestCenter.php:
```diff
@@ -0,0 +1,86 @@
+        return 'Fixes field name for ByPassProctor to a proper one (typo fix).';
+    }
+
+    public function up(Schema $schema): void
```
There is one more place where the mistype persists, the value is also stored in the config
https://github.com/oat-sa/extension-tao-testcenter/blob/v10.4.0/config/default/EligibilityCsvImporterFactory.conf.php#L37
Therefore it should be updated as well

vbyndych replied:
Could you clarify this? I'm not completely sure what has to happen here, as the place you've mentioned used constant which has already been updated.

augustas replied:
Right, on the first install of an extension, content from `EXTENSION_NAME/config/default/*` is copied to root `config/EXTENSION_NAME/*`. Since previously that constant had a wrong value, any instance with `taoTestCenter` extension already installed, would still have a mistake in config. It needs to be fixed in this migration as well

vbyndych replied:
Fixed. Thank you for your explanation. Every day I learn something new!

augustas commented on migrations/Version202307041134591871_taoTestCenter.php:
```diff
@@ -0,0 +1,86 @@
+        try {
+            $persistence = $this->getPersistence();
+
+            $query = sprintf(
```
No other migration ever executes queries directly on `statements` table. I don't have a better suggestion without making it complicated, but lets check that SQL is used for ontology

vbyndych replied:
I know it looks unusual, and I've asked folks if anybody could suggest a better approach, however, nobody faced something like that and the only suggestion was to leave it as is if it not hurts. But unfortunately, it is brakes the import, so I need to change it. 
As for the query itself, it is rather simple and ANSI SQL compatible so should work on any compatible engine (MySQL, PgSQL, MSSQL, etc.). Could you clarify which ontology SQL we should check?

augustas replied:
For example to check that `config/generis/ontology.conf.php` has `core_kernel_persistence_smoothsql_SmoothModel` registered as model. But yeah, I can't think of a concrete example where that would have been an issue, this migration won't run when migration to neo4j is already done

vbyndych replied:
Correct, this change is preparation for migration, so it has to run before neo4j migration. Unfortunately, I don't know what is the best way to ensure it, as fix and migration scripts in 2 different extensions.

vbyndych committed:
fix: RDF migration from sql to neo4j

- added config fix as value with typo was copied there.

vbyndych committed:
feat: RDF migration from sql to neo4j

- added check for main storage is SQL based, so I have no issue executing the query.
